contracts/interfaces/ISmartOrderStrategy.sol

```diff
@@ -16,8 +16,9 @@ interface ISmartOrderStrategy is IStrategy {
     struct Operation {
         address dest;
         address inputToken;
-        uint128 inputRatio;
-        uint128 dataOffset;
+        uint256 numerator;
+        uint256 denominator;
+        uint256 dataOffset;
```

Comment:
To save gas, replace a uint128 with a uint256. Using uint128 would incur read/write overhead and consume more gas.

Reply:
Can `numerator` and `denominator` be packed into uint256?

Reply:
They can be packed into a uint256, but this will consume more gas.

### Gas cost of the `executeStrategy` function when we declare `numerator` and `denominator` as uint256 separately:
- min: 23253
- avg: 126523
- median: 84289
- max: 610589

### Gas cost of the `executeStrategy` function when we pack `numerator` and `denominator` into a uint256:

- min: 23253
- avg: 126662
- median: 84450
- max: 610929